ops/aggregate.py

```diff
@@ -0,0 +1,889 @@
+        )
+
+        # Create montage
+        montage = utils.montage(cell_grid, shape=shape)
+        montages[channel_name] = montage
+      
+        # Save montage
+        output_path = os.path.join(output_dir, f'{output_prefix}_{channel_name}.tif')
+        
+        # Select channel if specified in channel_info
+        if channel_idx is not None:
+            montage = montage[channel_idx]
+        
+        save(output_path, montage,
+             display_mode='grayscale',
+             display_ranges=display_ranges[channel_name])
+        
+        print(f"Saved {channel_name} montage to {output_path}")    
+
+def create_sgrna_montage(df,
+                        gene,
+                        sgrna,
+                        channel,
+                        plotting_dict,
+                        population_feature='gene_symbol_0',
+                        sgrna_feature='sgRNA_0',
+                        num_cells=50,
+                        cell_size=40,
+                        shape=(5, 10),
+                        coordinate_cols=None):
+    """
+    Create a montage array for a specific gene-sgRNA-channel combination.
+    Designed to work with Snakemake workflows.
+    
+    Args:
+        df (pd.DataFrame): DataFrame with cell data
+        gene (str): Gene identifier to filter for
+        sgrna (str): sgRNA identifier to filter for
+        channel (str): Channel to create montage for
+        plotting_dict (dict): Dictionary mapping channels to filename info
+            Format: {'channel': {'filename': 'file_column', 'channel': idx}} or
+                   {'channel': {'filename': 'file_column'}}
+        population_feature (str): Column name for gene identifier
+        sgrna_feature (str): Column name for sgRNA identifier
+        num_cells (int): Number of cells to include
+        cell_size (int): Size of cell bounds box
+        shape (tuple): Shape of montage grid (rows, cols)
+        coordinate_cols (list): Names of coordinate columns for bounds
+    
+    Returns:
+        np.ndarray: Montage array ready for saving
+    """
+    if coordinate_cols is None:
+        coordinate_cols = ['i_0', 'j_0']
+        
+    # Filter for specific gene-sgRNA combination
+    mask = (df[population_feature] == gene) & (df[sgrna_feature] == sgrna)
+    df_subset = df[mask].copy()
+    
+    if len(df_subset) == 0:
+        return None
+    
+    # Take required number of cells
+    df_subset = df_subset.head(num_cells)    
+    df_subset = (df_subset
+                .pipe(annotate.add_rect_bounds, 
+                     width=cell_size, 
+                     ij=coordinate_cols, 
+                     bounds_col='bounds'))
+ 
+    # Parse channel info
+    channel_info = plotting_dict[channel]
+    if isinstance(channel_info, dict):
+        filename = channel_info['filename']
+        if filename == 'filename':
+            filename = 'filename'
+        else:
+            filename = f'filename_{filename}'
```

Comment:
```suggestion
        if filename != 'filename':
            filename = f'filename_{filename}'
```

Reply:
Does this maintain functionality? Maybe I interpreted the code wrong.

Reply:
Yup--changed!